include/net/nrf_cloud_pgps.h

```diff
@@ -315,12 +315,22 @@ void nrf_cloud_pgps_request_reset(void);
  * @param buf Pointer to data received from nRF Cloud.
  * @param buf_len Buffer size of data to be processed.
  *
- * @retval 0 A-GPS data successfully processed.
+ * @retval 0 P-GPS data successfully processed.
  * @retval -EFAULT An nRF Cloud P-GPS error code was processed.
  * @return A negative value indicates an error.
  */
 int nrf_cloud_pgps_process(const char *buf, size_t buf_len);
 
+/** @brief Processes binary P-GPS data using URL received from nRF Cloud.
+ *
+ * @param file_location Pointer to structure holding the hostname and filename.
+ *
+ * @retval 0 P-GPS data successfully processed.
+ * @retval -EFAULT An nRF Cloud P-GPS error code was processed.
+ * @return A negative value indicates an error.
+ */
+int nrf_cloud_pgps_update(struct nrf_cloud_pgps_result *file_location);
```

Comment:
It would be nice if handling of P-GPS data would be similar to REST, but now all three (MQTT, REST and CoAP) are handled somewhat differently.

If this can not be avoided, it should more clear what the application needs to do after calling `nrf_cloud_coap_pgps_data_get()`. Now, the only hint is that the output from that function is a `struct nrf_cloud_pgps_result` and `nrf_cloud_pgps_update()` takes in a pointer to such a struct. The name and the description of the function do not describe very well what the function does, I think.

Reply:
I will work on the comment to make it more clear.

If you look at the `do_pgps()` function in `main.c` in the sample commit in this PR, you'll see how it should be used.

Reply:
Thanks. I think the goal with APIs should be that it can be used without looking at a sample, but that may not always be so easy.

Reply:
Yes, I agree with the goal.

Reply:
I renamed `nrf_cloud_coap_pgps_data_get()` to `nrf_cloud_coap_pgps_url_get()` and updated various header comments and parameter names.